### DISCUSSION TITLE
ScrollIntoView method for DataGrid throws ArgumentOutOfRangeException

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#6099] [Q&A] [answered]
Hi,

I am trying to change the SelectedItem of a DataGrid from code behind but by default, the view doesn't scroll to the selected item. I see there is a `ScrollIntoView ` method available to do that. Using the below code throws an exception
`System.ArgumentOutOfRangeException: 'Specified argument was out of the range of valid values. (Parameter 'index')'`

Is there any other way to achieve this?

Code:
```
dataGrid.ScrollIntoView(dataGrid.SelectedItem, null);
```

**Comment by FoggyFinder** [comment]
Sounds like a bug to me. Can you add small test project?

**Reply by ShrutiJaiswal1494** [reply]
@FoggyFinder  you can download the test project zip from [here](https://drive.google.com/file/d/1G87SmD01GIDAAfPSiZfE-mjlS_N57szN/view?usp=sharing)

**Comment by ShrutiJaiswal1494** [comment, marked as answer]
Is there a way to do this? I am still not able to figure out what's wrong here.

**Reply by FoggyFinder** [reply]
probably you could workaround it like

```cs
private bool flag;
private void HandleSelectionChange(object sender, SelectionChangedEventArgs e)
{
    if (flag)
    {
        m_datagrid.ScrollIntoView(m_datagrid.SelectedItem, null);
    }
    else
    {
        flag = true;
    }
}
```

though I've created PR with a quick fix #6127

**Reply by ShrutiJaiswal1494** [reply]
Thank you. This works :)
